src/ui/components/shared/WorkspaceSettingsModal/WorkspaceSubscription.tsx

```diff
@@ -27,11 +25,33 @@ export const stripePromise = loadStripe(
 
 export default function WorkspaceSubscription({ workspaceId }: { workspaceId: string }) {
   const [view, setView] = useState<Views>("details");
-  const { data, loading } = hooks.useGetWorkspaceSubscription(workspaceId);
+  const [confirmed, setConfirmed] = useState(false);
+  const { data, loading, error, refetch } = hooks.useGetWorkspaceSubscription(workspaceId);
+
+  const subscription = data?.node.subscription;
+  const showTrialDetails =
+    subscription &&
+    subscription?.paymentMethods?.length === 0 &&
+    subscription.status === "trialing";
+
+  // clear the confirmed state if changing views
+  useEffect(() => {
+    if (confirmed && view !== "details") {
+      setConfirmed(false);
+    }
+  }, [confirmed, view]);
 
   if (loading) return null;
 
-  if (!data?.node.subscription) {
+  if (error) {
+    return (
+      <section className="space-y-8">
+        <p>Unable to load the subscription at this time</p>
+      </section>
```

Comment:
hmm, this could be tidier, but i think it's the right sentiment

> Unable to load the subscription at this time. We are looking into it on our end and feel free to email us at support@replay.io if this problem persists.

Reply:
we should use a `useEffect` + `pingTelemetry` to be notified of the error